tests/scripts/spack_manifest/test_getter.py

```diff
@@ -0,0 +1,480 @@
+    NoSectionComponentError,
+)
+
+class TestRootSpecGetter:
+    def test___init___valid_multi_target_spec(self):
+        manifest = {
+            "spack": {
+                "specs": ["$ROOT_SPEC"],
+                "definitions": [
+                    {
+                        "ROOT_PACKAGE": ["access-om2@git.1.0.0"]
+                    }
+                ]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+
+        assert root_spec_getter.root_spec == "access-om2@git.1.0.0", "Root spec should be correctly initialized from multi-target manifest."
+
+    def test___init___valid_single_target_spec(self):
+        manifest = {
+            "spack": {
+                "specs": ["access-om2@git.1.0.0"]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+
+        assert root_spec_getter.root_spec == "access-om2@git.1.0.0", "Root spec should be correctly initialized from single-target manifest."
+
+    def test___init___invalid_no_root_spec(self):
+        manifest = {
+            "spack": {
+                "specs": []
+            }
+        }
+
+        with pytest.raises(NoSectionError):
+            RootSpec(manifest)
+
+    def test___init___invalid_no_manifest(self):
+        with pytest.raises(NoSectionError):
+            RootSpec({})
+
+    def test_from_file__valid(self):
+        manifest_path = "tests/scripts/spack_manifest/injection/inputs/prerelease.spack.yaml"
+        root_spec_getter = RootSpec.from_file(manifest_path)
+
+        assert root_spec_getter.root_spec == "access-om2@git.2024.03.0=latest", "Root spec should be correctly retrieved from file."
+
+    def test_from_file__invalid(self):
+        manifest_path = "tests/scripts/spack_manifest/injection/inputs/invalid_manifest.yaml"
+
+        with pytest.raises(OSError):
+            RootSpec.from_file(manifest_path)
+
+    def test_get_name__valid(self):
+        manifest = {
+            "spack": {
+                "specs": ["access-om2@git.1.0.0"]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+        name = root_spec_getter.get_name()
+
+        assert name == "access-om2", "Root spec name should be correctly extracted."
+
+    def test_get_name__invalid(self):
+        manifest = {
+            "spack": {
+                "specs": ["@git.1.0.0"]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+        with pytest.raises(NoSectionComponentError):
+            root_spec_getter.get_name()
+
+    def test_get_ref__valid_git(self):
+        manifest = {
+            "spack": {
+                "specs": ["access-om2@git.1.0.0"]
+            }
+        }
+
+        root_spec_getter = RootSpec(manifest)
+        ref = root_spec_getter.get_ref()
+
+        assert ref == "1.0.0", "Root spec ref should be correctly extracted."
+
+    def test_get_ref__valid_non_git(self):
+        manifest = {
+            "spack": {
+                "specs": ["access-om2@1.0.0"]
```

Comment:
We don't impose any idea of what we think a version should look right? I think that's ok, but I wonder about people accidentally using a git ref with the `@git.` prefix.